mesonbuild/mdevenv.py

```diff
@@ -233,3 +242,5 @@ def run(options: argparse.Namespace) -> int:
         return e.returncode
     except FileNotFoundError:
         raise MesonException(f'Command not found: {args[0]}')
+    except KeyboardInterrupt:
+        pass
```

Comment:
What is the purpose of this?

Reply:
Avoid backtrace if you ctr-c to stop the program.

Reply:
So presumably this would be a problem before adding a --bin argument too...

Perhaps the answer here is actually that we shouldn't be using the subprocess module at all. We don't need python as a parent process performing any form of interceptions.

Reply:
> So presumably this would be a problem before adding a --bin argument too...

Yes, that's an old bug I fixed while testing this PR. I guess that could be a separate commit.

> Perhaps the answer here is actually that we shouldn't be using the subprocess module at all. We don't need python as a parent process performing any form of interceptions.

We have to wrap the command at least to setup devenv, not sure what you mean here.

Reply:
Made a separate commit.

Reply:
I didn't say meson devenv shouldn't exist -- I said that entering into the devenv command via subprocess seems like a bad idea. What I was thinking about was this instead:

https://github.com/eli-schwartz/meson/commit/a3a4963e64c2c836bead235153a11e9b70cc53ff

Reply:
@eli-schwartz oh I see, makes sense. Can you make that commit into a PR please?

Reply:
Removed my commit, that can be done in a different PR.